src/sdkClient/__tests__/clientAttributesDecoration.spec.ts

```diff
@@ -0,0 +1,270 @@
+
+    expect(Object.keys(client.getAttributes()).length).toEqual(0); // It should be zero after trying to add an invalid attribute
+
+    expect(client.clearAttributes()).toEqual(true);
+
+  });
+
+  test('Should return true if attributes map is valid', () => {
+    const validAttributes = {
+      'attributeKey1': 'attributeValue',
+      'attributeKey2': ['attribute', 'value'],
+      'attributeKey3': 25,
+      'attributeKey4': false
+    };
+
+    expect(client.setAttribute('attributeKey1', 'attributeValue')).toEqual(true); // It should be valid if the attribute value is a String, Number, Boolean or Lists.
+    expect(client.setAttribute('attributeKey2', ['attribute', 'value'])).toEqual(true); // It should be valid if the attribute value is a String, Number, Boolean or Lists.
+    expect(client.setAttribute('attributeKey3', 25)).toEqual(true); // It should be valid if the attribute value is a String, Number, Boolean or Lists.
+    expect(client.setAttribute('attributeKey4', false)).toEqual(true); // It should be valid if the attribute value is a String, Number, Boolean or Lists.
+    expect(client.setAttribute('attributeKey5', Date.now())).toEqual(true); // It should be valid if the attribute value is a String, Number, Boolean or Lists.
+
+    expect(client.removeAttribute('attributeKey5')).toEqual(true); // It should be capable of remove the attribute with that name
+    expect(client.getAttributes()).toEqual(validAttributes); // It should had stored every valid attributes.
+
+    expect(client.clearAttributes()).toEqual(true);
+
+    expect(client.setAttributes(validAttributes)).toEqual(true); // It should add them all because they are valid attributes.
+    expect(client.getAttributes()).toEqual(validAttributes); // It should had stored every valid attributes.
+
+    expect(client.clearAttributes()).toEqual(true);
+
+  });
+
+});
+
+describe('ATTRIBUTES DECORATION / evaluation', () => {
+
+  test('Evaluation attributes logic and precedence / getTreatment', () => {
+
+    // If the same attribute is “cached” and provided on the function, the value received on the function call takes precedence.
+    expect(client.getTreatment('split')).toEqual(undefined); // Nothing changes if no attributes were provided using the new api
+    expect(client.getTreatment('split', { func_attr_bool: true, func_attr_str: 'true' })).toEqual({ func_attr_bool: true, func_attr_str: 'true' }); // Nothing changes if no attributes were provided using the new api
+    expect(client.getAttributes()).toEqual({}); // Attributes in memory storage must be empty
+    client.setAttribute('func_attr_bool', false);
+    expect(client.getAttributes()).toEqual({ 'func_attr_bool': false }); // In memory attribute storage must have the unique stored attribute
+    expect(client.getTreatment('split', { func_attr_bool: true, func_attr_str: 'true' })).toEqual({ func_attr_bool: true, func_attr_str: 'true' }); // Function attributes has precedence against api ones
+    // @ts-ignore
```

Comment:
Emma, I think you can remove most `@ts-ignore` comments, like this one. Check which other ones can be removed